src/main/java/org/folio/helper/CheckinReceivePiecesHelper.java

```diff
@@ -216,15 +216,12 @@ private void checkIfAllPiecesFound(List<String> pieceIds, List<Piece> pieces) {
    * @return map passed as a parameter
    */
   protected Future<Map<String, List<Piece>>> storeUpdatedPieceRecords(Map<String, List<Piece>> piecesGroupedByPoLine, RequestContext requestContext) {
-    List<Future<Void>> futures = StreamEx
-      .ofValues(piecesGroupedByPoLine)
-      .flatMap(List::stream)
-      .filter(this::isSuccessfullyProcessedPiece)
-      .map(piece -> piece.withStatusUpdatedDate(new Date()))
-      .map(piece -> storeUpdatedPieceRecord(piece, requestContext))
-      .toList();
-
-    return GenericCompositeFuture.join(futures)
+    return GenericCompositeFuture.join(
+      extractAllPieces(piecesGroupedByPoLine)
+        .filter(this::isSuccessfullyProcessedPiece)
+        .map(piece -> piece.withStatusUpdatedDate(new Date()))
+        .map(piece -> storeUpdatedPieceRecord(piece, requestContext))
```

Comment:
we have duplicate call to this method https://github.com/folio-org/mod-orders/pull/947/files#diff-3a7f736956a9ecfff0816c2d0a92c1261471143f91a3a5ab53b1dcd448892117R81